### DISCUSSION TITLE
Implementing Glacier->Ocean coupling on multiple height levels

### DISCUSSION THREAD

**Discussion by Katetc** [ESCOMP/CISM-wrapper#89] [Show and tell]
Several people have worked on implementing the transfer of heat and salinity information from the ocean to the ice sheets in CESM in various ways. In this discussion, let's begin by pointing to the various code repositories with work that has been done and describing the state of that code (issues, age, work completed and work to be done, etc). Any documentation on this work, including relevant papers and presentations can be linked and discussed here as well. 

Our eventual goal is to support Antarctic ice sheet coupling on multiple depth levels with various ocean components in CESM3 and NorESM.

Those initially involved in this discussion are:
@Katetc @mvertens @billsacks @gunterl @whlipscomb @mpetrini-norce @hgoelzer 

Others are very welcome to join!

**Comment by whlipscomb** [comment]
Here are some google docs I found from 2019–2020, when several of us were first working on this coupling:

(1) Sarah's [notes](https://docs.google.com/document/d/1qp5bCMzNeAtIlSWiMLgrZk6JyP5ySM6lPrTIOBZvXYM/edit) on ice sheet–ocean coupling for LGM runs.

(2) Gunter's [notes](https://docs.google.com/document/d/1JxX5g1O6XPimyJkeDErOTnVrPxjxqs7hifOAtrmHjww/edit) on using ocean coupled fields to compute thermal forcing in CISM.

(3) My [notes](https://docs.google.com/document/d/1pY4ojeziRY7gE4Eqm36Rsnt1PtoIFnM1_c6fvts-ZLs/edit) on modifying CISM to ingest ocean T and S from the coupler.

I think all three docs should now be accessible to anyone with the link.

**Reply by micpetrini** [reply]
Here is another [Google Doc](https://docs.google.com/document/d/1scLtUcu5P77DSQ--01tCXt0hyG9eMgl8w8zG0bBClv4/edit?usp=sharing) I made at the very beginning of our POP-CISM coupling work (it includes several comments by Bill L.); it contains a detailed explanation of the separate vertical levels approach (seven levels, and seven mapping files), some thoughts on remapping (nearest-neighbour vs bilinear+extrapolation) and some paths/routines. It's an older document than 1)-2)-3) and probably outdated, but maybe still useful?

**Comment by micpetrini** [comment]
Here is the path on Derecho of the latest code version me and Sarah Bradley used to run tests (BG and T-compset) with POP-CISM coupling: /glade/u/home/mpetrini/mycesm2sandbox/cesm-2.1.1_cism_pop

**Reply by mvertens** [reply]
@mpetrini-norce - do you have any POP output files from these runs - that could be a starting point for a new data ocean mode?

**Reply by micpetrini** [reply]
I don't have outputs unfortunately - what I have is the scripts to:
(1) create first a fully coupled case with ocean forcing (and to save coupler history files):
/glade/u/home/mpetrini/cesm2cism2_oceancoup/create_PI-BG-pop.csh
(2) create a T-compset using coupler history files saved in (1) (both ocean and SMB)
/glade/u/home/mpetrini/cesm2cism2_oceancoup/create_PI-TG-pop.csh

Hope this can be somehow helpful.

**Comment by mvertens** [comment]
One option is to use the forced ocean/sea ice CESM2 simulation from [OMIP2](https://gmd.copernicus.org/articles/13/3643/2020/). 
The CESM OMIP2 output can be found at:
/glade/collections/cmip/CMIP6/OMIP/NCAR/CESM2/omip2/r1i1p1f1
The MISOMIP had an idealized domain and used MOM6 instead of POP.
